### PR TITLE
Added force deletion capability to log analytics module and updated requirements.txt

### PR DESCRIPTION
**Pull request by next-davidduquehernandez**
##### SUMMARY
Added force delete capability introduced on 05/26/2020 update. Updated requirement library 'azure-mgmt-loganalytics' to 1.0.0 and update its method calls.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_loganalyticsworkspace

##### ADDITIONAL INFORMATION
According to [azure documentation](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/delete-workspace) we can force log analytics deletion. This capability allow us to remove log analytics workspace permanently, cause default deletion has a soft-delete behavior.

As requirement was set to 0.2.0 (and this feature was introduced on 1.0.0), this forced me to update requirements and some sdk call methods.

This is my first pull request to an open source project. Please, tell me if I need to do some additional step or modify something in my code. Regards :)

Usage example:

Create log analytics workspace:
```
    - azure_rm_loganalyticsworkspace:
        name: loganalyticsex
        resource_group: rgtest01
        state: present
```

Create output:
```
ok: [localhost] => {
    "changed": false,
    "customer_id": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
    "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/rgtest01/providers/microsoft.operationalinsights/workspaces/loganalyticsex",
    "intelligence_packs": [
        {
            "enabled": false,
            "name": "CapacityPerformance"
        },
        {
            "enabled": false,
            "name": "AzureWebAppsAnalytics"
        },
        {
            "enabled": false,
            "name": "BehaviorAnalyticsInsights"
        },
        {
            "enabled": false,
            "name": "Security"
        },
        {
            "enabled": false,
            "name": "Updates"
        },
        {
            "enabled": false,
            "name": "AntiMalware"
        },
        {
            "enabled": true,
            "name": "LogManagement"
        },
        {
            "enabled": true,
            "name": "AzureResources"
        },
        {
            "enabled": false,
            "name": "ChangeTracking"
        },
        {
            "enabled": false,
            "name": "SQLAssessment"
        },
        {
            "enabled": false,
            "name": "SCOMAssessment"
        },
        {
            "enabled": false,
            "name": "ServiceDesk"
        },
        {
            "enabled": false,
            "name": "SQLAssessmentPlus"
        },
        {
            "enabled": false,
            "name": "ADAssessment"
        },
        {
            "enabled": false,
            "name": "AlertManagement"
        },
        {
            "enabled": false,
            "name": "AzureAutomation"
        },
        {
            "enabled": false,
            "name": "WireData"
        },
        {
            "enabled": false,
            "name": "SiteRecovery"
        },
        {
            "enabled": false,
            "name": "Backup"
        },
        {
            "enabled": false,
            "name": "SurfaceHub"
        },
        {
            "enabled": false,
            "name": "NetworkMonitoring"
        },
        {
            "enabled": false,
            "name": "Containers"
        },
        {
            "enabled": false,
            "name": "ContainerInsights"
        },
        {
            "enabled": false,
            "name": "ServiceMap"
        },
        {
            "enabled": false,
            "name": "AzureNetworking"
        },
        {
            "enabled": false,
            "name": "ADReplication"
        },
        {
            "enabled": false,
            "name": "Office365"
        },
        {
            "enabled": false,
            "name": "CompatibilityAssessment"
        },
        {
            "enabled": false,
            "name": "KeyVault"
        },
        {
            "enabled": false,
            "name": "ServiceFabric"
        },
        {
            "enabled": false,
            "name": "DnsAnalytics"
        },
        {
            "enabled": false,
            "name": "ApplicationInsights"
        },
        {
            "enabled": false,
            "name": "WireData2"
        },
        {
            "enabled": false,
            "name": "WaaSUpdateInsights"
        },
        {
            "enabled": false,
            "name": "AgentHealthAssessment"
        },
        {
            "enabled": false,
            "name": "AzureActivity"
        },
        {
            "enabled": false,
            "name": "HDInsight"
        },
        {
            "enabled": false,
            "name": "HDInsightKafka"
        },
        {
            "enabled": false,
            "name": "HDInsightSpark"
        },
        {
            "enabled": false,
            "name": "HDInsightHadoop"
        },
        {
            "enabled": false,
            "name": "AzureCdnCoreAnalytics"
        },
        {
            "enabled": false,
            "name": "HDInsightStorm"
        },
        {
            "enabled": false,
            "name": "HDInsightInteractiveQuery"
        },
        {
            "enabled": false,
            "name": "VMware"
        },
        {
            "enabled": false,
            "name": "SecurityCenterFree"
        },
        {
            "enabled": false,
            "name": "LogicAppsManagement"
        },
        {
            "enabled": false,
            "name": "LogicAppB2B"
        },
        {
            "enabled": false,
            "name": "AzureSQLAnalytics"
        },
        {
            "enabled": false,
            "name": "KeyVaultAnalytics"
        },
        {
            "enabled": false,
            "name": "AzureNSGAnalytics"
        },
        {
            "enabled": false,
            "name": "AzureAppGatewayAnalytics"
        },
        {
            "enabled": false,
            "name": "DeviceHealthProd"
        },
        {
            "enabled": false,
            "name": "WindowsDefenderATP"
        },
        {
            "enabled": false,
            "name": "WindowsDefenderATPStable"
        },
        {
            "enabled": false,
            "name": "InfrastructureInsights"
        },
        {
            "enabled": false,
            "name": "ProcessInvestigator"
        },
        {
            "enabled": false,
            "name": "Microsoft365Analytics"
        },
        {
            "enabled": false,
            "name": "ASRAnalytics"
        },
        {
            "enabled": false,
            "name": "AzureSentinelPrivatePreview"
        },
        {
            "enabled": false,
            "name": "SecurityInsights"
        },
        {
            "enabled": false,
            "name": "SecurityCenterNetworkTraffic"
        },
        {
            "enabled": false,
            "name": "WindowsFirewall"
        },
        {
            "enabled": false,
            "name": "WindowsEventForwarding"
        },
        {
            "enabled": false,
            "name": "InternalWindowsEvent"
        },
        {
            "enabled": false,
            "name": "CustomizedWindowsEventsFiltering"
        },
        {
            "enabled": false,
            "name": "WEFInternalUat"
        },
        {
            "enabled": false,
            "name": "WEFInternal"
        },
        {
            "enabled": false,
            "name": "WEF_10x"
        },
        {
            "enabled": false,
            "name": "WEF_10xDSRE"
        },
        {
            "enabled": false,
            "name": "DHCPActivity"
        },
        {
            "enabled": false,
            "name": "AzureDataFactoryAnalytics"
        },
        {
            "enabled": false,
            "name": "AzureSecurityOfThings"
        },
        {
            "enabled": false,
            "name": "SQLVulnerabilityAssessment"
        },
        {
            "enabled": false,
            "name": "SQLAdvancedThreatProtection"
        },
        {
            "enabled": false,
            "name": "VMInsights"
        },
        {
            "enabled": false,
            "name": "WinLog"
        },
        {
            "enabled": false,
            "name": "NetFlow"
        }
    ],
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "force": null,
            "intelligence_packs": null,
            "location": null,
            "name": "loganalyticsex",
            "password": null,
            "profile": null,
            "resource_group": "rgtest01",
            "retention_in_days": null,
            "secret": null,
            "sku": "per_gb2018",
            "state": "present",
            "subscription_id": null,
            "tags": null,
            "tenant": null
        }
    },
    "location": "westeurope",
    "management_groups": [],
    "name": "loganalyticsex",
    "provisioning_state": "Succeeded",
    "public_network_access_for_ingestion": "Enabled",
    "public_network_access_for_query": "Enabled",
    "retention_in_days": 30,
    "shared_keys": {
        "primary_shared_key": "mh360coZP3tUvaD6RG57ABnuOdkchgTq+UsrMNhjwAjFIljnB+qsrAvfKMSCkBV+D4rpNkbHZG/xDM1QDKBSfQ==",
        "secondary_shared_key": "NziIgGDqsHn7tY+acDxKPG7XTlCCRMI0cS6sUc+fA5M7GCZvcClaaweTY6eaOmIzCapw4nIWfnDLYvg9B1I+qQ=="
    },
    "sku": "pergb2018",
    "tags": {},
    "type": "Microsoft.OperationalInsights/workspaces",
    "usages": [
        {
            "current_value": 0.0,
            "limit": -1.0,
            "name": {
                "localized_value": "Data Analyzed",
                "value": "DataAnalyzed"
            },
            "next_reset_time": "2020-09-25T20:00:00.000Z",
            "quota_period": "P1D",
            "unit": "Bytes"
        }
    ],
    "workspace_capping": {
        "daily_quota_gb": -1.0,
        "data_ingestion_status": "RespectQuota",
        "quota_next_reset_time": "Fri, 25 Sep 2020 20:00:00 GMT"
    }
}
```

Delete log analytics workspace with force:
```
    - azure_rm_loganalyticsworkspace:
        name: loganalyticsex
        resource_group: rgtest01
        state: absent
        force: true
```

Delete output:
```
changed: [localhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "force": true,
            "intelligence_packs": null,
            "location": null,
            "name": "loganalyticsex",
            "password": null,
            "profile": null,
            "resource_group": "rgtest01",
            "retention_in_days": null,
            "secret": null,
            "sku": "per_gb2018",
            "state": "absent",
            "subscription_id": null,
            "tags": null,
            "tenant": null
        }
    }
}
```

### CODE REVIEW

**Comment by imjoseangel** [comment]
Hey @next-davidduquehernandez as a good startup point, it is important to open an issue explaining what do you want to achieve and then link the PR with the comment `Fixes #yourissuenumber`. Document the issue with the requirements to understand what we should expect in the PR.

Thank you!

File: plugins/modules/azure_rm_loganalyticsworkspace.py [hunk 1]
```diff
@@ -63,6 +63,10 @@
             - Disable one pack by setting it to C(false). For example "Backup:false".
             - Other intelligence packs not list in this property will not be changed.
         type: dict
+    force:
+        description:
+            - Deletes the workspace without the recovery option. A workspace that was deleted with this flag cannot be recovered.
+        type: bool
```
Comment (imjoseangel) [inline plugins/modules/azure_rm_loganalyticsworkspace.py:70]: `default: no`

File: plugins/modules/azure_rm_loganalyticsworkspace.py [hunk 2]
```diff
@@ -173,7 +177,8 @@ def __init__(self):
             location=dict(type='str'),
             sku=dict(type='str', default='per_gb2018', choices=['free', 'standard', 'premium', 'unlimited', 'per_node', 'per_gb2018', 'standalone']),
             retention_in_days=dict(type='int'),
-            intelligence_packs=dict(type='dict')
+            intelligence_packs=dict(type='dict'),
+            force=dict(type='bool')
```
Comment (imjoseangel) [inline plugins/modules/azure_rm_loganalyticsworkspace.py]: `force=dict(type='bool', default=False)`

**Review by imjoseangel — CHANGES_REQUESTED** [review]
@next-davidduquehernandez I have added some comments to add a default value to the force option. Remember to open an issue with the description of what do you want to fix. Nice Job!

**Comment by next-davidduquehernandez** [comment]
Fixes #274

File: plugins/modules/azure_rm_loganalyticsworkspace.py (hunk 1 above)
Comment (next-davidduquehernandez) [reply plugins/modules/azure_rm_loganalyticsworkspace.py:70]: Added default: false better than 'no'

**Comment by next-davidduquehernandez** [comment]
Thank you for your comments @imjoseangel. I introduced some tests to this PR and fixed 'azure_rm_loganalyticsworkspace_info' too. I has created issue #274 and tried to link with this PR, but I don't know if it's already done.

**Comment by imjoseangel** [comment]
LGTM

**Comment by Fred-sun** [comment]
RROR: Found 12 pep8 issue(s) which need to be resolved:
ERROR: plugins/modules/azure_rm_subnet.py:267:161: E501: line too long (261 > 160 characters)
ERROR: plugins/modules/azure_rm_subnet.py:267:262: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_subnet.py:268:1: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_subnet.py:268:161: E501: line too long (256 > 160 characters)
ERROR: plugins/modules/azure_rm_subnet.py:268:257: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_subnet.py:269:1: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_subnet.py:269:161: E501: line too long (271 > 160 characters)
ERROR: plugins/modules/azure_rm_subnet.py:269:272: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_subnet.py:270:1: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_subnet.py:270:161: E501: line too long (271 > 160 characters)
ERROR: plugins/modules/azure_rm_subnet.py:270:272: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_subnet.py:271:1: E128: continuation line under-indented for visual indent
RROR: plugins/modules/azure_rm_subnet.py:0:0: doc-choices-do-not-match-spec: Argument 'serviceName' in argument_spec found in delegations defines choices as (['Microsoft.Web/serverFarms', 'Microsoft.ContainerInstance/containerGroups', 'Microsoft.Netapp/volumes', 'Microsoft.HardwareSecurityModules/dedicatedHSMs', 'Microsoft.ServiceFabricMesh/networks', 'Microsoft.Logic/integrationServiceEnvironments', 'Microsoft.Batch/batchAccounts', 'Microsoft.Sql/managedInstances', 'Microsoft.Web/hostingEnvironments', 'Microsoft.BareMetal/CrayServers', 'Microsoft.BareMetal/MonitoringServers', 'Microsoft.Databricks/workspaces', 'Microsoft.BareMetal/AzureHostedService', 'Microsoft.BareMetal/AzureVMware', 'Microsoft.BareMetal/AzureHPC', 'Microsoft.BareMetal/AzurePaymentHSM', 'Microsoft.StreamAnalytics/streamingJobs', 'Microsoft.DBforPostgreSQL/serversv2', 'Microsoft.AzureCosmosDB/clusters', 'Microsoft.MachineLearningServices/workspaces', 'Microsoft.DBforPostgreSQL/singleServers', 'Microsoft.DBforPostgreSQL/flexibleServers', 'Microsoft.DBforMySQL/serversv2', 'Microsoft.DBforMySQL/flexibleServers', 'Microsoft.ApiManagement/service', 'Microsoft.Synapse/workspaces', 'Microsoft.PowerPlatform/vnetaccesslinks', 'Microsoft.Network/managedResolvers', 'Microsoft.Kusto/clusters']) but documentation defines choices as ([])
ERROR: plugins/modules/azure_rm_subnet.py:0:0: doc-elements-mismatch: Argument 'delegations' in argument_spec specifies elements as dict,but elements is not documented

**Comment by next-davidduquehernandez** [comment]
> RROR: Found 12 pep8 issue(s) which need to be resolved:
> ERROR: plugins/modules/azure_rm_subnet.py:267:161: E501: line too long (261 > 160 characters)
> ERROR: plugins/modules/azure_rm_subnet.py:267:262: W291: trailing whitespace
> ERROR: plugins/modules/azure_rm_subnet.py:268:1: E128: continuation line under-indented for visual indent
> ERROR: plugins/modules/azure_rm_subnet.py:268:161: E501: line too long (256 > 160 characters)
> ERROR: plugins/modules/azure_rm_subnet.py:268:257: W291: trailing whitespace
> ERROR: plugins/modules/azure_rm_subnet.py:269:1: E128: continuation line under-indented for visual indent
> ERROR: plugins/modules/azure_rm_subnet.py:269:161: E501: line too long (271 > 160 characters)
> ERROR: plugins/modules/azure_rm_subnet.py:269:272: W291: trailing whitespace
> ERROR: plugins/modules/azure_rm_subnet.py:270:1: E128: continuation line under-indented for visual indent
> ERROR: plugins/modules/azure_rm_subnet.py:270:161: E501: line too long (271 > 160 characters)
> ERROR: plugins/modules/azure_rm_subnet.py:270:272: W291: trailing whitespace
> ERROR: plugins/modules/azure_rm_subnet.py:271:1: E128: continuation line under-indented for visual indent
> RROR: plugins/modules/azure_rm_subnet.py:0:0: doc-choices-do-not-match-spec: Argument 'serviceName' in argument_spec found in delegations defines choices as (['Microsoft.Web/serverFarms', 'Microsoft.ContainerInstance/containerGroups', 'Microsoft.Netapp/volumes', 'Microsoft.HardwareSecurityModules/dedicatedHSMs', 'Microsoft.ServiceFabricMesh/networks', 'Microsoft.Logic/integrationServiceEnvironments', 'Microsoft.Batch/batchAccounts', 'Microsoft.Sql/managedInstances', 'Microsoft.Web/hostingEnvironments', 'Microsoft.BareMetal/CrayServers', 'Microsoft.BareMetal/MonitoringServers', 'Microsoft.Databricks/workspaces', 'Microsoft.BareMetal/AzureHostedService', 'Microsoft.BareMetal/AzureVMware', 'Microsoft.BareMetal/AzureHPC', 'Microsoft.BareMetal/AzurePaymentHSM', 'Microsoft.StreamAnalytics/streamingJobs', 'Microsoft.DBforPostgreSQL/serversv2', 'Microsoft.AzureCosmosDB/clusters', 'Microsoft.MachineLearningServices/workspaces', 'Microsoft.DBforPostgreSQL/singleServers', 'Microsoft.DBforPostgreSQL/flexibleServers', 'Microsoft.DBforMySQL/serversv2', 'Microsoft.DBforMySQL/flexibleServers', 'Microsoft.ApiManagement/service', 'Microsoft.Synapse/workspaces', 'Microsoft.PowerPlatform/vnetaccesslinks', 'Microsoft.Network/managedResolvers', 'Microsoft.Kusto/clusters']) but documentation defines choices as ([])
> ERROR: plugins/modules/azure_rm_subnet.py:0:0: doc-elements-mismatch: Argument 'delegations' in argument_spec specifies elements as dict,but elements is not documented

Hi @Fred-sun, thanks for your comment but I think that you was trying to write it to this PR https://github.com/ansible-collections/azure/pull/264 and not mine, cause I'm not modifying plugins/modules/azure_rm_subnet.py, could be?

Regards :)

**Comment by Fred-sun** [comment]
Sorry,  This is my mistake!

**Comment by Fred-sun** [comment]
@next-davidduquehernandez Please add line "azure_rm_loganalyticsworkspace" to  pr-pipelines.yml. Thank you very much!

File: tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      Containers: true
+    retention_in_days: "{{ retention_days }}"
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- assert:
+    that:
+      - output.changed
+      - output.retention_in_days == retention_days
+      - output.intelligence_packs | json_query('[?name == `Backup`].enabled') | first == true
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml]:
```suggestion
      #- output.intelligence_packs | json_query('[?name == `Backup`].enabled') | first == true
```

File: tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      - output.intelligence_packs | json_query('[?name == `Containers`].enabled') | first == true
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml]:
```suggestion
      #- output.intelligence_packs | json_query('[?name == `Containers`].enabled') | first == true
```

File: tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    show_usages: true
+  register: facts
+
+- assert:
+    that:
+      - not facts.changed
+      - facts.workspaces | length > 0
+      - facts.workspaces[0].retention_in_days == retention_days
+      - facts.workspaces[0].intelligence_packs | length > 0
+      - facts.workspaces[0].intelligence_packs | json_query('[?name == `Backup`].enabled') | first == true
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml]:
```suggestion
     #- facts.workspaces[0].intelligence_packs | json_query('[?name == `Backup`].enabled') | first == true
```

File: tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      - facts.workspaces[0].intelligence_packs | json_query('[?name == `Containers`].enabled') | first == true
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_loganalyticsworkspace/tasks/main.yml]:
```suggestion
      #- facts.workspaces[0].intelligence_packs | json_query('[?name == `Containers`].enabled') | first == true
```

**Review by Fred-sun — COMMENTED** [review]
Currently, python3 in the pipeline lacks the filter json_query. Cancel it temporarily and change it back after updating the environment. Thank you!

**Comment by next-davidduquehernandez** [comment]
Added "azure_rm_loganalyticsworkspace" to pr-pipelines.yml and commented lines with json_query. :)

**Pull request merged**